packages/amplify-cli-core/src/feature-flags/featureFlagProvider.ts

```diff
@@ -0,0 +1,422 @@
+import Ajv, { AdditionalPropertiesParams } from 'ajv';
+import * as fs from 'fs-extra';
+import _ from 'lodash';
+import * as path from 'path';
+import { JSONSchema7, JSONSchema7Definition } from 'json-schema';
+import { CLIEnvironmentProvider, JSONUtilities, JSONValidationError } from '..';
+import { FeatureFlags, FeatureFlagRegistration, FeatureFlagsEntry, FeatureFlagType } from '.';
+import { AmplifyConfigFileName } from '../constants';
+import { FeatureFlagFileProvider } from './featureFlagFileProvider';
+import { FeatureFlagEnvironmentProvider } from './featureFlagEnvironmentProvider';
+
+export class FeatureFlagProvider {
```

Comment:
I would name this class `FeatureFlags`. IMO Providers are single methods that take 0 args and "provide" some output